src/ui/MissingTerm/MissingTerms.ts

```diff
@@ -0,0 +1,154 @@
+    /**
+     * Whether to allow the end-user to click a missing term to re-inject it in the query as
+     * an exact phrase match (i.e., as an expression between double quote characters).
+     *
+     * **Default:** `true`
+     */
+    clickable: ComponentOptions.buildBooleanOption({ defaultValue: true }),
+    /**
+     * The text to display before missing terms.
+     *
+     * **Default:** The localized string for `Missing`.
+     */
+    caption: ComponentOptions.buildLocalizedStringOption({ defaultValue: 'Missing' })
+  };
+
+  static doExport = () => {
+    exportGlobally({
+      MissingTerms: MissingTerms
+    });
+  };
+
+  /* Used to split terms and phrases. Match characters that can separate words or caracter for chinese, japanese and korean.
+  * Han: Unicode script for chinesse caracter
+  * We only need to import 1 asian script because what is important here is the space between the caracter and any script will contain it
+  */
+  private wordBoundary = '(([\\p{Han}])?([^(\\p{Latin}-)])|^|$)';
+
+  /**
+   * Creates a new `MissingTerms` component instance.
+   * @param element The element on which to instantiate the component.
+   * @param options The configuration options for the component.
+   * @param bindings The bindings required by the component to function normally. If not set, these will be automatically resolved (with a slower execution time).
+   * @param result The query result item to associate the component with.
+   */
+  constructor(
+    public element: HTMLElement,
+    public options?: IMissingTermsOptions,
+    bindings?: IComponentBindings,
+    public result?: IQueryResult
+  ) {
+    super(element, MissingTerms.ID, bindings);
+
+    this.options = ComponentOptions.initComponentOptions(element, MissingTerms, options);
+    this.addMissingTerms();
+  }
+  /**
+   *Returns all original basic query expression terms that were not matched by the result item the component instance is associated with.
+   */
+  public get missingTerms(): string[] {
+    return this.result.absentTerms.filter(term => {
+      const regex = this.createRegex(false, term);
+      return regex.test(this.queryStateModel.get('q'));
+    });
+  }
+  /**
+   * Re-injects a term as an exact phrase match expression in the query.
+   */
+  public includeTermInQuery(term: string) {
+    if (this.missingTerms.indexOf(term) === -1) {
+      this.logger.warn('the term to re-inject is not present in the missing terms');
+      return;
+    }
+
+    let newQuery: string = this.queryStateModel.get('q');
+    const regex = this.createRegex(true, term);
+    let stillhasResults = true;
+    while (stillhasResults) {
+      const results = regex.exec(newQuery);
+      stillhasResults = results !== null;
+      if (stillhasResults) {
+        const offset = results[0].indexOf(term);
+        newQuery = [newQuery.slice(0, results.index + offset), '"', term, '"', newQuery.slice(results.index + term.length + offset)].join(
```

Comment:
To answer @olamothe suggestion to use `String.replace`, since my regex needs to check left and right to see if it is a complete word it select them. So I will always lose the character before and after the term, and since the characters are never the same, I can't simply add them with the second parameter of replace.